src/sentry/issues/escalating.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime, timedelta
+from typing import List, Tuple
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Limit,
+    Offset,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry.models import Group
+from sentry.utils.json import JSONData
+from sentry.utils.snuba import raw_snql_query
+
+
+def extract_project_and_group_ids(groups: List[Group]) -> Tuple[List[int], List[int]]:
+    """Return all project and group IDs from a list of Group"""
+    group_ids = []
+    project_ids = []
+    for group in groups:
+        project_ids.append(group.project_id)
+        group_ids.append(group.id)
+
+    # This also removes duplicated project ids
+    return list(set(project_ids)), group_ids
+
+
+def query_groups_past_counts(groups: List[Group]) -> JSONData:
+    project_ids, group_ids = extract_project_and_group_ids(groups)
+    group_id_col = Column("group_id")
+    proj_id_col = Column("project_id")
+
+    query = Query(
+        match=Entity("events"),
+        select=[
+            proj_id_col,
+            group_id_col,
+            Function("toStartOfHour", [Column("timestamp")], "hourBucket"),
+            Function("count", []),
+        ],
+        groupby=[proj_id_col, group_id_col, Column("hourBucket")],
+        where=[
+            Condition(proj_id_col, Op.IN, Function("tuple", project_ids)),
+            Condition(Column("group_id"), Op.IN, Function("tuple", group_ids)),
+            Condition(Column("timestamp"), Op.GTE, datetime.now() - timedelta(days=14)),
+            Condition(Column("timestamp"), Op.LT, datetime.today()),
```

Comment:
@onewland do `datetime.now() - timedelta(days=14)` and `datetime.today()` look good to you?

Reply:
I think you can pass these values but I'm not sure if you're going to get the buckets back that you're hoping for. Here's what I mean. As I write this comment it's 

`2023-03-31 14:37:02.728976`. If you do `timedelta(days=14)` then that date is going to be `2023-03-17 14:37:02.728976`

and `toStartOfHour('2023-03-17 14:37:02.728976')` will come out to `2023-03-17 14:00:00.000000` so you'll get every hour between `2023-03-17 14:00:00.000000` and `2023-03-31 14:00:00.000000`. If that works for the forecast then it's no problem but I imagine that you might want your days to start on `00:00:00.000000`/